specs/applicationdata/ApplicationData.md

```diff
@@ -0,0 +1,432 @@
+
+        /// Delete the specified settings container, its subcontainers, and all settings in the hierarchy.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdatacontainer.deletecontainer
+        void DeleteContainer(String name);
+    };
+
+    /// Provides access to the application data store.
+    /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata
+    [contract(ApplicationDataContract, 1)]
+    runtimeclass ApplicationData : Windows.Foundation.IClosable
+    {
+        /// Get an instance of ApplicationData for the current user.
+        /// @note Requires the current process has package identity.
+        /// @note This is equivalent to Windows.Storage.ApplicationData.Current if running in an AppContainer.
+        /// @note This is equivalent to Windows.Storage.ApplicationDataManager.CreateForPackageFamily(GetCurrentPackageFamilyName()) if not running in an AppContainer.
+        static ApplicationData GetDefault();
+
+        /// Get an instance of ApplicationData for the specified user.
+        /// @note This is equivalent to Windows.Storage.ApplicationDataManager.GetForUserAsync()
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.getforuserasync
+        static ApplicationData GetForUser(Windows.System.User user);
+
+        /// Get an instance of ApplicationData for the specified package family for the current user.
+        /// @note This is equivalent to Windows.Storage.ApplicationDataManager.CreateForPackageFamily()
+        /// @see https://learn.microsoft.com/uwp/api/windows.management.core.applicationdatamanager.createforpackagefamily
+        static ApplicationData GetForPackageFamily(String packageFamilyName);
+
+        /// Get an instance of ApplicationData for the specified unpackaged app for the current user.
+        static ApplicationData GetForUnpackaged(String publisher, String name);
+
+        /// Return true if the package family supports the machine data store.
+        Boolean IsMachinePathSupported();
+
+        /// Return the path for the local cache data store not included in backup and restore operations.
+        /// @note This is equivalent to Windows.Storage.ApplicationDataManager.LocalCacheFolder().Path()
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.localcachefolder
+        String LocalCachePath { get; };
+
+        /// Return the path for the local data store. This location is backed up to the cloud.
+        /// @note This is equivalent to Windows.Storage.ApplicationDataManager.LocalFolder().Path()
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.localfolder
+        String LocalPath { get; };
+
+        /// Return the path for the machine data store.
+        /// @note This location is per-machine (not per-user like other *Path properties).
+        String MachinePath { get; };
```

Comment:
What happens if you ask for MachinePath and MachineFolder when IsMachinePathSupported is false?

Reply:
If IsMachinePathSupported()=false
* MachinePath returns empty string (std::wstring void ctor)
* MachineFolder returns null

In the spirit of "GetPrinters()" on a system that doesn't support printers, you get back a valid empty result. Likewise, MachinePath/Folder give you a valid empty result you can't do more with.

Reply:
Document - of course.

And yes, consistent with SharedLocalFolder and other "null"-esque experiences.